### PR TITLE
[JS] Improve Error Page

### PR DESCRIPTION
**Pull request by TakLee96**
connects #102 
connects #132

### CODE REVIEW

File: content/js/handling-errors.md [hunk 1]
```diff
@@ -2,25 +2,47 @@ All asynchronous methods in `skygear` return [Promise](https://www.promisejs.org
...
-skygear.signup('johndoe', 'john.doe@example.com', 'verysecurepassword').then(() => {
+``` javascript
+skygear.signupWithUsername('johndoe', 'verysecurepassword').then(() => {
   // user logged in successfully
 }, (error) => {
-  if (error.code == skygear.Error.DUPLICATED) {
-    console.log('username has been taken');
-  } else {
-    console.log('unexpected error', error.code)
+  if (error.code === 109) {
```
Comment (bensonby) [inline content/js/handling-errors.md]: should not use "magic number" here

File: content/js/handling-errors.md [hunk 2]
```diff
@@ -2,25 +2,47 @@ All asynchronous methods in `skygear` return [Promise](https://www.promisejs.org
...
+    alert('username or email already exists');
```
Comment (bensonby) [inline content/js/handling-errors.md]: I believe all code examples use `console` for logging? you may want to use `console.error` for this case?

File: content/js/handling-errors.md [hunk 3]
```diff
@@ -2,25 +2,47 @@ All asynchronous methods in `skygear` return [Promise](https://www.promisejs.org
...
 
 ```
 
 ## Ignoring errors
 
-```js
+``` javascript
 // omit the error callback
-skygear.signup('johndoe', 'john.doe@example.com', 'verysecurepassword').then(() => {
+skygear.signupWithUsername('johndoe', 'verysecurepassword').then(() => {
```
Comment (bensonby) [inline content/js/handling-errors.md]: I am not sure about the intention here for "ignoring errors". I don't think we should ignore a rejection from a promise

File: content/js/handling-errors.md (hunk 1 above)
Comment (TakLee96) [inline content/js/handling-errors.md]: Let's wait for @rickmak 's explanation on accessing predefined error codes

File: content/js/handling-errors.md [hunk 4]
```diff
@@ -2,25 +2,40 @@ All asynchronous methods in `skygear` return [Promise](https://www.promisejs.org
...
+import error from 'skygear/lib/error';
+
+skygear.signupWithUsername('johndoe', 'verysecurepassword').then(() => {
   // user logged in successfully
 }, (error) => {
-  if (error.code == skygear.Error.DUPLICATED) {
-    console.log('username has been taken');
-  } else {
-    console.log('unexpected error', error.code)
+  if (error.code === error.Duplicated) {
```
Comment (bensonby) [inline content/js/handling-errors.md]: two variables both named `error`...

Comment (TakLee96) [inline content/js/handling-errors.md]: Oops

**Pull request merged**